lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -29,16 +29,19 @@ def print_results(toolinfo: dict, with_option: bool) -> bool:
         """
         printed = False
         for group, gval in sorted(toolinfo.items()):
-            for host, tools in sorted(gval.items()):
+            for host, hostitems in sorted(gval.items()):
+                label = toolinfo[group][host]["label"]
+                host_string = f"host: {host}" + (f", label: {label}" if label else "")
+                tools = hostitems["tools"]
                 if tools:
                     if not with_option:
-                        s = ", ".join(sorted(tools.keys()))
+                        tool_string = ", ".join(sorted(tools.keys()))
                     else:
                         tools_with_options = [
                             f"{t} {' '.join(v)}" for t, v in sorted(tools.items())
                         ]
-                        s = ", ".join(tools_with_options)
-                    print(f"group: {group}; host: {host}; tools: {s}")
+                        tool_string = ", ".join(tools_with_options)
```

Comment:
There is no need to instantiate the list that we're assigning to `tools_with_options`, here -- what we want is the generator expression which would create the list -- that we can pass to the `join()` call at line 43:
```python
                        tools_with_options = (
                            f"{t} {' '.join(v)}" for t, v in sorted(tools.items())
                        )
                        tool_string = ", ".join(tools_with_options)
```
Replacing the square brackets with parentheses creates a _generator_ rather than a tuple, in this context.  We then hand that generator to `join()` which iterates over the generated items (as it would the items from the list), without actually creating the list.